### PR TITLE
test: server and cypress support mock logins with different roles

### PR DESCRIPTION
**Pull request by matthieu-foucault**
Required for e2e tests in #32 and #639

### CODE REVIEW

**Review by marcellmueller — APPROVED** [review]
Thanks for doing this, looks great. 

Do you think a small section in the readme explaining how this works would be helpful?

**Review by marcellmueller — CHANGES_REQUESTED** [review]
Oh it looks like the test is failing CI

File: docs/auth.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
+# Authentication and Authorization
```
Comment (marcellmueller) [inline docs/auth.md:1]: Thanks for this :heart:

**Review by marcellmueller — APPROVED** [review]
Nice job and I appreciate the readme and eliminating a lot of the `cy.wait` throughout the tests.

I noticed a few in some tests still such as `upload.spec.js`, I assume those are still needed? `cy.contains` doesn't help in every scenario?

**Comment by matthieu-foucault** [comment]
> I noticed a few in some tests still such as upload.spec.js, I assume those are still needed? cy.contains doesn't help in every scenario?

They may not be needed anymore. I did a search and replace of the `cy.get().contains(` pattern, so I didn't look into the `upload.spec.js` too much

Some `wait`s on the initial page load are still needed, as we have "detached DOM" issues otherwise, caused by the may renders that occur on page load

**Pull request merged**